Packages/Netherlands3D/WebServiceReader/Runtime/Scripts/Layer/WMSImageLayer.cs

```diff
@@ -87,26 +119,44 @@ private Tile CreateNewTile(Vector2Int tileKey)
             tile.gameObject.layer = tile.gameObject.transform.parent.gameObject.layer;
             Vector2Int origin = new Vector2Int(tileKey.x+(tileSize/2), tileKey.y + (tileSize / 2));
             tile.gameObject.transform.position = CoordConvert.RDtoUnity(origin);
-            
+            var sizeVector = new Vector3(tileSize, tileSize, tileSize);
+
+            if (prefabIsDecalProjector)
+            {
+                tile.gameObject.GetComponent<DecalProjector>().size = sizeVector;
+
+            }
+            else
+            {
+                tile.gameObject.transform.localScale = sizeVector;
+            }
+
             return tile;
         }
 
         IEnumerator DownloadTexture(TileChange tileChange, Action<TileChange> callback = null)
         {
             var tileKey = new Vector2Int(tileChange.X, tileChange.Y);
+
+            if (!tiles.ContainsKey(tileKey))
+            {
+                Debug.Log("Tile key does not exist", this.gameObject);
+                yield break;
+            }
+
+            Tile tile = tiles[tileKey];
```

Comment:
Moet hier niet nog een nullcheck op? In deze methode wordt RemoveGameObjectFromTile aangeroepen zonder dat deze key wordt ge-remove'd; wat betekent dat hier een null uit kan komen